x-pack/plugins/security_solution/public/management/pages/endpoint_hosts/view/details/components/endpoint_details_tabs.tsx

```diff
@@ -27,34 +28,52 @@ interface EndpointDetailsTabs {
   id: string;
   name: string;
   content: JSX.Element;
+  route: string;
 }
 
+const EndpointDetailsTab = memo(
+  ({
+    tab,
+    isSelected,
+    handleTabClick,
+  }: {
+    tab: EndpointDetailsTabs;
+    isSelected: boolean;
+    handleTabClick: () => void;
+  }) => {
+    const setUrl = useNavigateByRouterEventHandler(tab.route);
+    const onClick = useCallback(
+      (e: React.MouseEvent<HTMLAnchorElement | HTMLButtonElement, MouseEvent>) => {
+        setUrl(e);
+        handleTabClick();
+      },
+      [setUrl, handleTabClick]
+    );
```

Comment:
I don't think you need all of this. `useNavigateByRouterEventHandler` also accepts a callback, so you can just (I think):

```suggestion
const onClick = useNavigateByRouterEventHandler(tab.route, handleTabClick);
```

Reply:
Indeed. That's even better. Thanks. 🙏

Reply:
done 8253ccdd9e4bcb06de841758ad2166f4a2c0b779